src/js/tracker.js

```diff
@@ -1604,25 +1563,70 @@
 				helpers.addEventListener(windowAlias, 'resize', activityHandler);
 				helpers.addEventListener(windowAlias, 'focus', activityHandler);
 				helpers.addEventListener(windowAlias, 'blur', activityHandler);
+			}
 
+			if (activityTrackingConfig.enabled && (resetActivityTrackingOnPageView || installingActivityTracking)) {
 				// Periodic check for activity.
 				lastActivityTime = now.getTime();
-				clearInterval(pagePingInterval);
-				pagePingInterval = setInterval(function heartBeat() {
-					var now = new Date();
-
-					// There was activity during the heart beat period;
-					// on average, this is going to overstate the visitDuration by configHeartBeatTimer/2
-					if ((lastActivityTime + configHeartBeatTimer) > now.getTime()) {
-						// Send ping if minimum visit time has elapsed
-						if (configMinimumVisitTime < now.getTime()) {
-							logPagePing(finalizeContexts(context, contextCallback)); // Grab the min/max globals
-						}
+
+				for (var key in activityTrackingConfig.configurations) {
```

Comment:
You could use `let` instead of `var` to keep it scoped to the block.

Reply:
Agreed. However, the whole codebase is for(var in ...). Leaving as is for consistency but we should look at updating these, particularly for v3.